### PR TITLE
[Config] Print contents of loaded config file

### PR DESCRIPTION
**Pull request by illusionyy**

### CODE REVIEW

File: src/xenia/config.cc [hunk 1]
```diff
@@ -25,13 +25,20 @@ std::shared_ptr<cpptoml::table> ParseFile(
...
+  char bom[3] = {0};
   file.read(bom, sizeof(bom));
   if (file.fail() || bom[0] != '\xEF' || bom[1] != '\xBB' || bom[2] != '\xBF') {
     file.clear();
     file.seekg(0);
   }
 
+  XELOGI("Loading config: {}", xe::path_to_utf8(filename));
+  std::string line{};
+  // dump contrnts of loaded config file
```
Comment (Margen67) [inline src/xenia/config.cc]:
```suggestion
  // dump contents of loaded config file
```

**Comment by Margen67** [comment]
image

**Comment by illusionyy** [comment]
works for me, didn't test games
[xenia.log](https://github.com/xenia-canary/xenia-canary/files/12152846/xenia.log)

**Comment by illusionyy** [comment]
should be fixed @Margen67

File: src/xenia/config.cc [hunk 2]
```diff
@@ -25,12 +25,21 @@ std::shared_ptr<cpptoml::table> ParseFile(
                                    " could not be opened for parsing");
   }
   // since cpptoml can't parse files with a UTF-8 BOM we need to skip them
-  char bom[3];
+  char bom[3]{};
```
Comment (Margen67) [inline src/xenia/config.cc:28]: Was this intentional?

File: src/xenia/config.cc [hunk 3]
```diff
@@ -25,12 +25,21 @@ std::shared_ptr<cpptoml::table> ParseFile(
...
   file.read(bom, sizeof(bom));
   if (file.fail() || bom[0] != '\xEF' || bom[1] != '\xBB' || bom[2] != '\xBF') {
     file.clear();
     file.seekg(0);
   }
+  if (xe::endsWith(filename.string(), ".config.toml")) {
```
Comment (Margen67) [inline src/xenia/config.cc]: Did you try just checking if it was null or whatever?
```c++
  if (filename) {
```

File: src/xenia/config.cc (hunk 2 above)
Comment (illusionyy) [reply src/xenia/config.cc:28]: Yes, it's to default initialize it

File: src/xenia/config.cc (hunk 3 above)
Comment (illusionyy) [reply src/xenia/config.cc]: If `.size()` in `filename` is 0, it won't pass this condition anyway

Comment (Margen67) [reply src/xenia/config.cc]: I mean can you check if it's null instead of doing all this?

Comment (illusionyy) [reply src/xenia/config.cc]: It only freezes when it tries to list the contents of patch files. I added this check to only print stuff from the config

**Comment by illusionyy** [comment]
Waitng for C++20

**Comment by illusionyy** [comment]
Maybe not

**Pull request merged**